Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -96,63 +96,51 @@ - (void)realTrack:(NSString *)event properties:(NSDictionary *)properties integr
-            SEGLog(@"[[AMPRevenue revenue] setProductIdentifier:%@];", productId);
-        }
-        
-        //Receipt is meant to be of type NSData
-        id receipt = [properties objectForKey:@"receipt"];
-        if (receipt && [receipt isKindOfClass:[NSString class]] && ![receipt isEqualToString:@""]) {
-            [self.amprevenue setReceipt:receipt];
-            SEGLog(@"[[AMPRevenue revenue] setReceipt:%@];", receipt);
-        }
-        id revenueType = [properties objectForKey:@"revenueType"] ?: [properties objectForKey:@"revenue_type"];
-        if (revenueType && [revenueType isKindOfClass:[NSString class]] && ![revenueType isEqualToString:@""]) {
-            [self.amprevenue setRevenueType:revenueType];
-            SEGLog(@"[AMPRevenue revenue] setRevenueType:%@];", revenueType);
-        }
-        NSLog(@"Price : %@, Quantity : %@", price, quantity);
-        [self.amplitude logRevenueV2:self.amprevenue];
-        SEGLog(@"[Amplitude logRevenueV2:%@];", self.amprevenue);
-        
-    } else {
-        // fallback to logRevenue v1
-        id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
-        if (!productId || ![productId isKindOfClass:[NSString class]]) {
-            productId = nil;
-        }
-        id quantity = [properties objectForKey:@"quantity"];
-        if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
-            quantity = [NSNumber numberWithInt:1];
-        }
-        id receipt = [properties objectForKey:@"receipt"];
-        if (!receipt || ![receipt isKindOfClass:[NSString class]]) {
-            receipt = nil;
-        }
-        [self.amplitude logRevenue:productId
-                          quantity:[quantity integerValue]
-                             price:revenueOrTotal
-                           receipt:receipt];
-        SEGLog(@"[Amplitude logRevenue:%@ quantity:%d price:%@ receipt:%@];", productId, [quantity integerValue], revenueOrTotal, receipt);
+        [self trackLogRevenueV2:properties andRevenueOrTotal:revenueOrTotal];
+        return;
+    }
+
+    // fallback to logRevenue v1
+    id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"] ?: nil;
+    id quantity = [properties objectForKey:@"quantity"] ?: [NSNumber numberWithInt:1];
+    id receipt = [properties objectForKey:@"receipt"] ?: nil;
+    [self.amplitude logRevenue:productId
+                      quantity:[quantity integerValue]
+                         price:revenueOrTotal
+                       receipt:receipt];
+    SEGLog(@"[Amplitude logRevenue:%@ quantity:%d price:%@ receipt:%@];", productId, [quantity integerValue], revenueOrTotal, receipt);
+}
+
+- (void)trackLogRevenueV2:(NSDictionary *)properties andRevenueOrTotal:(NSNumber *)revenueOrTotal
+{
+    id price = [properties objectForKey:@"price"] ?: revenueOrTotal;
+    id quantity = [properties objectForKey:@"quantity"] ?: [NSNumber numberWithInt:1];
```

Comment:
In Android, we set quantity to 1 if "price" is null but either revenue or total has a value. That's because Amplitude will calculate their internal revenue with price * quantity. Maybe we should guard against this in iOS as well: https://github.com/segment-integrations/analytics-android-integration-amplitude/blob/master/src/main/java/com/segment/analytics/android/integrations/amplitude/AmplitudeIntegration.java#L203-L211